### PR TITLE
Get and update sync code only after sync chain is created

### PR DESCRIPTION
**Pull request by darkdh**
Otherwise, we end up caching wrong sync code when page is loaded and it
will only be updated after a page refresh

Resolves https://github.com/brave/brave-browser/issues/10857

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. STR in https://github.com/brave/brave-browser/issues/10857
2. Reset sync chain
3. Create a new sync chain and remember the sync code
4. After sync chain created, click `View Sync Code`
5. Code in step 3 & 4 should be the same

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by darkdh** [comment]
restart CI for window only

File: browser/resources/settings/brave_sync_page/brave_sync_configure.js [hunk 1]
```diff
@@ -82,11 +78,20 @@ Polymer({
     return displayDate.toDateString()
   },
 
-  onViewSyncCode_: function() {
+  getSyncCode_: async function() {
+    if (!!this.syncCode)
```
Comment (bsclifton) [inline browser/resources/settings/brave_sync_page/brave_sync_configure.js:82]: might be useful to add a comment like "only set `syncCode` if it's not already set

**Comment by petemill** [comment]
@darkdh was this issue caused because the 'configure' component is not removed and re-attached when finishing setting up the new sync chain and exiting the sync section?
If there's still a timing issue with this fix in the future, the proper fix is probably to have the syncCode bubble down from the main sync page component.

**Comment by darkdh** [comment]
>> @darkdh was this issue caused because the 'configure' component is not removed and re-attached when finishing setting up the new sync chain and exiting the sync section?
If there's still a timing issue with this fix in the future, the proper fix is probably to have the syncCode bubble down from the main sync page component.
No, it is actually caused by user navigate to sync page for the first time without any sync chain setup. At that point, it tries to get sync code from pref but it shouldn't

File: browser/resources/settings/brave_sync_page/brave_sync_configure.js (hunk 1 above)
Comment (bsclifton) [reply browser/resources/settings/brave_sync_page/brave_sync_configure.js:82]: per our call - this needs to be cleared when you leave the chain

Comment (darkdh) [reply browser/resources/settings/brave_sync_page/brave_sync_configure.js:86]: fixed

File: browser/resources/settings/brave_sync_page/brave_sync_configure.js [hunk 2]
```diff
@@ -82,11 +78,20 @@ Polymer({
     return displayDate.toDateString()
   },
 
-  onViewSyncCode_: function() {
+  getSyncCode_: async function() {
```
[the remark this answers is not in the document]
Comment (darkdh) [reply browser/resources/settings/brave_sync_page/brave_sync_configure.js]: fixed

**Comment by btlechowski** [comment]
LGTM

Verification passed on

Brave | 1.13.50 Chromium: 84.0.4147.105 (Official Build) nightly (64-bit)
-- | --
Revision | a6b12dfad6663f13a7e16e9a42a6a4975374096b-refs/branch-heads/4147@{#943}
OS | Ubuntu 18.04 LTS

Verified test plan from https://github.com/brave/brave-core/pull/6182#issue-455400985.
Verified test case from https://github.com/brave/brave-browser/issues/10857#issuecomment-662589445

**Pull request merged**